skunkworks_crow/src/main/java/org/odk/share/views/ui/main/MainActivity.java

```diff
@@ -245,20 +251,39 @@ private void addListItemDivider() {
     @Override
     public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
         super.onRequestPermissionsResult(requestCode, permissions, grantResults);
-
         switch (requestCode) {
             case STORAGE_PERMISSION_REQUEST_CODE:
                 if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                     setUpLoader();
                 } else {
-                    //close the app if the permission is denied
-                    finish();
+                    if ((Build.VERSION.SDK_INT >= Build.VERSION_CODES.M &&
```

Comment:
Even this version check is redundant because every time the version is going to be greater than `M`, because you request only for the greater versions than `M`

Reply:
But method `shouldShowRequestPermissionRationale` has a compiler check for API > 23, if we remove the API check, the code cannot pass the compiler.

Reply:
Oops, I forgot that ok for now lets keep it.